.travis.yml

```diff
@@ -5,14 +5,20 @@ os:
   - linux
   - osx
 
+dist: bionic
+
 addons:
   apt:
-    packages: lcov
+    packages: 
+      - cppcheck
+      - lcov
+
 
 before_install:
-  - cd test && sudo chmod u+x setup.sh && ./setup.sh
+  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then HOMEBREW_NO_AUTO_UPDATE=1 brew install cppcheck; fi
+  - cd test && sudo chmod u+x lib/cppchecksetup.sh && sudo chmod u+x lib/googletestsetup.sh && ./lib/cppchecksetup.sh "CI" && ./lib/googletestsetup.sh
```

Comment:
Why do you call the setup scripts here? I though they are to be called by `Makefile` if the corresponding library is not already set up.

Reply:
The google test one was called there before-hand, so I chose to keep it consistent. The only thing that second line really does is complain if cppcheck isn't installed on Travis for whatever reason. I think the idea was to do a manual setup before calling make incase we needed to do something different on Travis. But, we don't yet (not sure if we'll ever), so should I just remove it for now?

Reply:
I think removing is a good option. We want to use the travis-native cppcheck anyway, right? Installing this manually unnecessarily increases travis run.

Reply:
Oh sorry, by manual setup, I meant for anything else we might want, not for installing cppcheck again. That being said, there's no harm in removing that line and we can always add it back later, so I've done that for now. Only thing to keep in mind is that now, if cppcheck fails to install on Travis, it'll untar the tar, rather than error instantly.

Edit: Apparently, there's a reason why the gtest setup line was there that went over my head. It error'd on Travis complaining about [using a C++ compiler instead of a C one](https://travis-ci.org/Hyp-ed/hyped-2020/jobs/620752514?utm_medium=notification&utm_source=github_status) 

I'm not really sure what to do about this, so I'll leave this line here for now.

Reply:
Would it be okay to resolve this like this for now, and maybe deal with this later?